packages/web/app/features/Invoice/InvoiceForm/InvoiceItem.jsx

```diff
@@ -91,12 +94,51 @@ export const InvoiceItemRow = ({
       productName: value.productName,
       productCode: value.code,
       productDiscountable: value.discountable,
+      // Store nested product details so downstream logic can read price list info
+      product: {
+        name: value.productName,
+        code: value.code,
+        discountable: value.discountable,
+      },
     });
   };
 
   const onClick = () => {
     setIsExpanded(!isExpanded);
   };
+  // Todo: Determine input state based on productPriceManualEntry when it's implemented
+  const priceListPrice = item.product?.invoicePriceListItem?.price;
+  const hasKnownPrice = Boolean(priceListPrice);
+  const hidePriceInput =
+    (priceListPrice !== null && priceListPrice !== undefined) ||
+    !editable ||
+    fetchedPriceData?.price;
+
+  // Todo: Also need to lookup the insurance plan for the product
+  const { data: fetchedPriceData, isFetching } = useInvoicePriceListItemPriceQuery({
+    encounterId,
+    productId: item.productId,
+    enabled: !hasKnownPrice,
+    onSuccess: data => {
+      if (!data?.price) {
+        return;
+      }
+      // If there is a price list price, update the form data
+      const { price } = data;
+
+      const nextProduct = {
+        ...(item.product || {}),
+        invoicePriceListItem: {
+          ...(item.product?.invoicePriceListItem || {}),
+          price,
+        },
+      };
+      formArrayMethods.replace(index, {
```

Comment:
this updates the form data and then the components that display prices always read the data from the form values. I was on the fence about whether to do it this way or to conditionally read the prices from the data OR the form values but I ended up going with this way.